kong/clustering/wrpc_control_plane.lua

```diff
@@ -195,30 +208,17 @@ function _M:handle_cp_websocket()
     dp_version = dp_version,
     log_suffix = log_suffix,
     basic_info = nil,
-    basic_info_semaphore = semaphore.new()
+    basic_info_semaphore = semaphore.new(),
+    dp_plugins_map = empty_table,
+    cp_ref = self,
+    config_hash = string.rep("0", 32), -- initial hash
+    sync_status = CLUSTERING_SYNC_STATUS.UNKNOWN,
   }
-  self.clients[w_peer.conn] = client
-  w_peer:spawn_threads()
+  w_peer.client = client
+  w_peer.cp = self
```

Comment:
This allows the `ReportMetadata` handler to access the client object, so the handling of `client:update_sync_status` can be called there.